pkg/collector/agonsticapi/loader.go

```diff
@@ -0,0 +1,90 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package agonsticapi
+
+/*
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <dlfcn.h>
+#include "loader.h"
+
+extern void* open_library(char *library, const char **error)
+{
+	// load the library
+	void *lib_handle = dlopen(library, RTLD_LAZY);
+	if (!lib_handle) {
+		*error = strdup("unable to open shared library");
+		return NULL;
+	}
+	return lib_handle;
+}
+*/
+import "C"
+
+import (
+	"errors"
+	"fmt"
+	"unsafe"
+
+	"github.com/DataDog/datadog-agent/comp/core/autodiscovery/integration"
+	tagger "github.com/DataDog/datadog-agent/comp/core/tagger/def"
+	integrations "github.com/DataDog/datadog-agent/comp/logs/integrations/def"
+	"github.com/DataDog/datadog-agent/pkg/aggregator/sender"
+	"github.com/DataDog/datadog-agent/pkg/collector/check"
+	"github.com/DataDog/datadog-agent/pkg/util/option"
+)
+
+const loaderName string = "agnosticapi"
+
+type agonsticAPILoader struct {
+	tagger tagger.Component
+}
+
+func newLoader(_ sender.SenderManager, _ option.Option[integrations.Component], tagger tagger.Component) (check.Loader, error) {
+	return &agonsticAPILoader{
+		tagger: tagger,
+	}, nil
+}
+
+// Name returns Shared Library loader name
+func (*agonsticAPILoader) Name() string {
+	return loaderName
+}
+
+func (sl *agonsticAPILoader) String() string {
+	return "Agnostic API Loader"
+}
+
+// Load returns a Shared Library check
+func (sl *agonsticAPILoader) Load(senderManager sender.SenderManager, config integration.Config, instance integration.Data) (check.Check, error) {
+	var cErr *C.char
+
+	name := "lib" + config.Name + ".dylib"
```

Comment:
Not for a PoC but we will need to extend things to non-Darwin systems at some point.